.github/workflows/check-iana-registries.yml

```diff
@@ -0,0 +1,53 @@
+name: Check for IANA special-purpose address registry updates
+
+on:
+  schedule:
+    - cron: "20 16 * * *"
+  workflow_dispatch:
+
+jobs:
+  check-iana-registries:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: write
+      pull-requests: write
+
+    steps:
+      - name: Checkout iana/data from main branch
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: iana/data
+
+      # If the branch already exists, this will fail, which will remind us about
+      # the outstanding PR.
+      - name: Create an iana-registries-gha branch
+        run: |
+            git checkout --track origin/main -b iana-registries-gha
+
+      - name: Retrieve the IANA special-purpose address registries
+        run: |
+            IANA_IPV4="https://www.iana.org/assignments/iana-ipv4-special-registry/iana-ipv4-special-registry-1.csv"
+            IANA_IPV6="https://www.iana.org/assignments/iana-ipv6-special-registry/iana-ipv6-special-registry-1.csv"
+
+            REPO_IPV4="iana/data/iana-ipv4-special-registry-1.csv"
+            REPO_IPV6="iana/data/iana-ipv6-special-registry-1.csv"
+
+            curl --output "${REPO_IPV4}" "${IANA_IPV4}"
+            curl --output "${REPO_IPV6}" "${IANA_IPV6}"
```

Comment:
I think this could end up with PRs that contain empty CSV files. `curl` exits `0` on HTTP 4XX/5XX.

You can get non-zero exit and a nice verbose response by adding the following `--fail`, keep progress bars out of the logs (but still show errors) with `--silent --show-error` and `--location` can be added to follow redirects.

Reply:
Here's a new test Action run and PR with the new `curl` flags added:
* https://github.com/jprenken/boulder/actions/runs/15864784541/job/44729515701
* https://github.com/jprenken/boulder/pull/4